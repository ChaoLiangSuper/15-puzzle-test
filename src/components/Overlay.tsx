import React from 'react';
import { IOverlayProps } from '../types';

const Overlay: React.FC<IOverlayProps> = ({ displayText }) => (
  <div className='overlay'>{displayText}</div>
);

export default Overlay;
