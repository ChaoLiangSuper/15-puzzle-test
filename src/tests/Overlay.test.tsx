import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Overlay from '../components/Overlay';

describe('<Overlay />', () => {
  it('should initialize without crash and match snapshot', () => {
    const exampleText = 'TEST STRING';
    const wrapper: ShallowWrapper = shallow(
      <Overlay displayText={exampleText} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toBe(exampleText);
  });
});
