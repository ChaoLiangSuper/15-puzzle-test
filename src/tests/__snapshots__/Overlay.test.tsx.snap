// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay /> should initialize without crash and match snapshot 1`] = `
<div
  className="overlay"
>
  TEST STRING
</div>
`;
